RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitWorkerMultiFit.cc

```diff
@@ -128,7 +135,14 @@ EcalUncalibRecHitWorkerMultiFit::set(const edm::EventSetup& es)
         // for the time correction methods
         es.get<EcalTimeBiasCorrectionsRcd>().get(timeCorrBias_);
 
-        int nnoise = noisecovariances->EBG12SamplesCorrelation.size();
+        int nnoise = SampleVector::RowsAtCompileTime;
+        SampleMatrix &noisecorEBg12 = noisecors_[1][0];
```

Comment:
Ok, yes this is maybe confusing.  The code here creates a non-const reference &noisecorEBg12 to a particular element of noisecors_, and likewise for the other elements, which are then all initialized in the existing code at
https://github.com/bendavid/cmssw/blob/f4837b2132cd150cdaf934fadf7e5ae07f3ff004/RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitWorkerMultiFit.cc#L146-L156